feat: support rabbitmq headers

DaAlbrecht opened the pull request:
## Purpose

This pull request adds the ability to include custom headers in RabbitMQ message properties. It introduces support for the headers field in BasicProperties and updates the Java code to map these headers during message publishing

## Examples

```bal
import ballerinax/rabbitmq;

rabbitmq:ConnectionConfiguration config = {
    username: "guest",
    password: "guest"
};
rabbitmq:Client rabbitmqClient = check new ("localhost", 5672, config);

public function main() returns error? {
    map<anydata> headers = {"header1": "value1", "header2": "value2"};
    rabbitmq:BasicProperties basicProperties = {headers: headers};
    rabbitmq:AnydataMessage anydataMessage = {routingKey: "test", content: "hello world", properties: basicProperties};
    check rabbitmqClient->publishMessage(anydataMessage);
}
```

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [x] Checked native-image compatibility

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ballerina-platform/module-ballerinax-rabbitmq?pullRequest=997) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ballerina-platform/module-ballerinax-rabbitmq?pullRequest=997) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ballerina-platform/module-ballerinax-rabbitmq?pullRequest=997) it.</sub>

aashikam commented:
Hi, Could you add a test for this?

DaAlbrecht commented:
Hi @aashikam 

The tests didn't pass because I haven't added support for consuming messages and the corresponding headers. Fixed this now and added a new test.

tharindu-nw commented:
Some compiler plugin tests seem to be failing still

aashikam commented:
<img width="1019" alt="Screenshot 2024-09-27 at 21 28 29" src="https://github.com/user-attachments/assets/d0bda69a-259b-4fc0-b454-37a4b9cb1c5e">

I inspected the error. This comes for the following files but they have correctly used the `rabbitmq:AnydataMessage` record. 
<img width="661" alt="Screenshot 2024-09-27 at 21 31 49" src="https://github.com/user-attachments/assets/798b95cc-9693-4446-8cf1-ee06f01e26ad">

And the PR does not include any compiler plugin related changes.

DaAlbrecht commented:
Thank you for the fix! @aashikam  merged your pr

aashikam reviewed (approved):
LGTM

aashikam commented on ballerina/rabbitmq_commons.bal:
```diff
@@ -43,11 +43,13 @@ public const TOPIC_EXCHANGE = "topic";
 # + contentType - The content type of the message
 # + contentEncoding - The content encoding of the message
 # + correlationId - The client-specific ID that can be used to mark or identify messages between clients
+# + headers - A map of additional arbitrary headers to be included in the message.
```
```suggestion
# + headers - A map of additional arbitrary headers to be included in the message
```
